public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Seg.tsx

```diff
@@ -90,41 +96,70 @@ const SelReload = ({ loadOptions, allowCustomValue, onChange, onClose }: SelRelo
 };
 
 type SelSingleLoadProps = {
+  value: string;
   loadOptions: (filter: string) => Promise<SelVal[]>;
   onClose: () => void;
-  onChange: (v: SelVal) => void;
+  onChange: (v: SelVal | null) => void;
   allowCustomValue?: boolean;
+  isClearable: boolean;
 };
 
-const SelSingleLoad = ({ loadOptions, allowCustomValue, onChange, onClose }: SelSingleLoadProps): JSX.Element => {
+const SelSingleLoad = ({
+  value,
+  loadOptions,
+  allowCustomValue,
+  isClearable,
+  onChange,
+  onClose,
+}: SelSingleLoadProps): JSX.Element => {
   const [loadState, doLoad] = useAsyncFn(loadOptions, [loadOptions]);
 
   useEffect(() => {
     doLoad();
   }, [doLoad, loadOptions]);
 
+  const options = loadState.value ?? [];
+
+  // if the current-value is not in the list-of-options, and it is not empty, we have to add
+  // it to the options
+  const valueInOptions = options.some((x) => x.value === value);
```

Comment:
I'm not sure this is always the case:

https://user-images.githubusercontent.com/1170767/118804701-efa66300-b89c-11eb-9ffa-397089f537a6.mov

Reply:
oh, nice catch :) yeah, i forgot i hacked the [+] buttons by doing a segment with a value="+". i fixed it (i added "support" for placeholder text).